src/Microsoft.SqlTools.Migration/MigrationService.cs

```diff
@@ -607,21 +607,24 @@ public void InitializeService(IProtocolEndpoint serviceHost)
         /// </summary>
         internal async Task HandleGetArmTemplateRequest(
     string skuRecommendationReportFilePath,
-    RequestContext<string> requestContext)
+    RequestContext<List<string>> requestContext)
         {
             try
             {
                 ProvisioningScriptServiceProvider provider = new ProvisioningScriptServiceProvider();
                 List<SkuRecommendationResult> recommendations = ExtractSkuRecommendationReportAction.ExtractSkuRecommendationsFromReport(skuRecommendationReportFilePath);
-                SqlArmTemplate template = provider.GenerateProvisioningScript(recommendations);
-
-                string jsonOutput = JsonConvert.SerializeObject(
-                template,
-                Formatting.Indented,
-                new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore, Culture = CultureInfo.InvariantCulture }
-                );
-
-                await requestContext.SendResult(jsonOutput);
+                List<SqlArmTemplate> templateList = provider.GenerateProvisioningScript(recommendations);
+                List<string> armTemplates = new List<string>();
+                foreach (SqlArmTemplate template in templateList)
```

Comment:
Should we check if templaList is undefined or it is handled?

Reply:
please add a null check here, nice catch Stuti!

Reply:
I don't think it would be null or undefined , it can be empty though.